src/Android/MainActivity.cs

```diff
@@ -278,6 +297,40 @@ private void ListenYubiKey(bool listen)
             }
         }
 
+        private async Task<bool> ListenU2F(Dictionary<string, object> data)
+        {
+            var appId = Uri.Parse((string)data["appId"]);
+            var challenge = Encoding.ASCII.GetBytes((string) data["challenge"]);
```

Comment:
I believe that challenge is a "url safe" base 64 encoded string. See here for a method that can decode that: https://github.com/bitwarden/mobile/blob/master/src/Core/Services/TokenService.cs#L203

Might want to mvoe that method into a static core helper.

Reply:
Thanks for the hint, will try that in the next few days.

I'll clean that up once I got everything up and running. I just noticed that I'll need at least 2 entrypoints to the U2F functionality when I started and before that I just picked up where you left comments about implementing U2F sometime in the future.

Reply:
Do you by any chance know how the U2F appId must be formatted? I fear that this is my actual problem here. The Bitwarden Server returns a link to the https://vault.bitwarden.com/app-id.json and from the documentation of U2F I read, that's not a valid app identifier. Instead it seems like this just tells me which appIds will be accepted by the server when I open that reference.

I think I might need to insert the Android App ID in the appId field, but I'm not sure how to obtain that yet.

Reply:
The app id should be `https://vault.bitwarden.com/app-id.json`.

If that doesn't work, then maybe try `android:apk-key-hash:dUGFzUzf3lmHSLBDBIv+WaFyZMI`, which is defined in that JSON file for the Bitwarden android app.